### PR TITLE
Tooltip update for artwork price range tooltip

### PR DESCRIPTION
**Pull request by oxaudo**
Trying to follow your patterns for Artwork bricks(https://github.com/artsy/force/pull/2428/files) I'm introducing ReactTooltip here that should replace current tooltip class.

Couple of notes inline. This still has several issues which I bring up below. And for reference - here is what it looks locally right now. The `?` for Artist Marketdata below the image is rendered directly from reaction as part of rendering `MarketInsights`. The one next to price is rendered from price jade file for artwork component.

scroll

### CODE REVIEW

File: src/desktop/apps/artwork/components/commercial/templates/price.jade [hunk 1]
```diff
@@ -8,11 +8,9 @@ if artwork.sale_message || isPermanentCollection
         .artwork-commercial__price-label Price
         .artwork-commerical__price
           | #{artwork.sale_message}
-          span(
-            class= artwork.is_price_range ? 'help-tooltip' : ''
-            data-message= 'The range is an approximate indication of the work’s price point, and the exact price is quoted upon request.'
-            data-anchor= 'bottom-right'
-          )
+          if artwork.is_price_range
+            .react-tooltip-container
```
[the remark this answers is not in the document]
Comment (damassi) [reply src/desktop/apps/artwork/components/commercial/templates/price.jade]: Totally makes sense 👍 the idea is to fit stuff into existing layout

File: src/desktop/apps/artwork/components/commercial/templates/price.jade [hunk 2]
```diff
@@ -8,11 +8,9 @@ if artwork.sale_message || isPermanentCollection
...
+              != stitch.components.Tooltip({message: 'The range is an approximate indication of the work’s price point, and the exact price is quoted upon request.', hoverAlign: 'left' })
```
[the remark this answers is not in the document]
Comment (damassi) [reply src/desktop/apps/artwork/components/commercial/templates/price.jade]: I would try this with `yarn watch` running against a local version of reaction and see how it goes

File: src/desktop/lib/global_react_modules.js [hunk 3]
```diff
@@ -1,6 +1,16 @@
 import React from 'react'
 import { Artwork as _Artwork } from '@artsy/reaction/dist/Components/Artwork'
 import { ArtworkGrid as _ArtworkGrid } from '@artsy/reaction/dist/Components/ArtworkGrid'
+import { Tooltip as ReactionTooltip } from '@artsy/reaction/dist/Components/Tooltip'
+import { Help as ReactionHelpIcon } from '@artsy/reaction/dist/Assets/Icons/Help'
```
Comment (damassi) [inline src/desktop/lib/global_react_modules.js]: This file currently contains some WIP stuff around the grid, hence the reason why I have a few components imported and then wrapped as an additional export -- since you're not making any modifications before data is being passed to reaction, can do 

```javascript
export { Tooltip } from '@artsy/reaction/dist/Components/Tooltip'
export { Help } from '@artsy/reaction/dist/Assets/Icons/Help'
```

File: src/desktop/lib/global_react_modules.js [hunk 4]
```diff
@@ -1,6 +1,16 @@
...
+
+export const Tooltip = props => {
+  return (
+    <ReactionTooltip {...props.props}>
+      <ReactionHelpIcon />
+    </ReactionTooltip>
+  )
+}
```
[the remark this answers is not in the document]
Comment (damassi) [reply src/desktop/lib/global_react_modules.js]: Reaction should contain the canonical representation / look / feel and not be impacted by force environment, so it def makes sense to define the font there and later, if more configuration is needed a prop added to the component.

File: src/desktop/lib/global_react_modules.js [hunk 5]
```diff
@@ -1,6 +1,16 @@
 import React from 'react'
 import { Artwork as _Artwork } from '@artsy/reaction/dist/Components/Artwork'
 import { ArtworkGrid as _ArtworkGrid } from '@artsy/reaction/dist/Components/ArtworkGrid'
+import { Tooltip as ReactionTooltip } from '@artsy/reaction/dist/Components/Tooltip'
+import { Help as ReactionHelpIcon } from '@artsy/reaction/dist/Assets/Icons/Help'
+
+export const Tooltip = props => {
+  return (
+    <ReactionTooltip {...props.props}>
```
[the remark this answers is not in the document]
Comment (damassi) [reply src/desktop/lib/global_react_modules.js]: Following the comment above about `export { Tooltip } from ...` and eliminating wrappers, the component as its used in Force should match exactly with how its used in reaction

File: src/desktop/lib/global_react_modules.js [hunk 6]
```diff
@@ -1,6 +1,16 @@
 import React from 'react'
 import { Artwork as _Artwork } from '@artsy/reaction/dist/Components/Artwork'
 import { ArtworkGrid as _ArtworkGrid } from '@artsy/reaction/dist/Components/ArtworkGrid'
+import { Tooltip as ReactionTooltip } from '@artsy/reaction/dist/Components/Tooltip'
```
[the remark this answers is not in the document]
Comment (damassi) [reply src/desktop/lib/global_react_modules.js]: Yup, this references the package as it's loaded in from NPM unless you're using the `yarn watch` flow which in turn references a local copy

File: src/desktop/apps/artwork/components/commercial/templates/price.jade (hunk 2 above)
[the remark this answers is not in the document]
Comment (damassi) [reply src/desktop/apps/artwork/components/commercial/templates/price.jade]: Since we can't do multiline in jade might be good to move the `hoverAlign: 'left'` prop to the front so its not lost to future devs behind the long `message` string.

File: src/desktop/lib/global_react_modules.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (alloy) [reply src/desktop/lib/global_react_modules.js]: What @damassi says 👍

**Comment by oxaudo** [comment]
With the last update here the question mark looks correct when using `yarn watch`. In real life that should wait for reaction deploy before merging, right @damassi?

**Comment by damassi** [comment]
@oxaudo -- yup! in Reaction we now support [automated deployments](https://github.com/artsy/reaction#commits-and-deployments) so once that's out this should be good to merge 👍

File: src/desktop/lib/global_react_modules.js [hunk 7]
```diff
@@ -1,6 +1,16 @@
 import React from 'react'
 import { Artwork as _Artwork } from '@artsy/reaction/dist/Components/Artwork'
 import { ArtworkGrid as _ArtworkGrid } from '@artsy/reaction/dist/Components/ArtworkGrid'
+import { Tooltip } from '@artsy/reaction/dist/Components/Tooltip'
+import { Help } from '@artsy/reaction/dist/Assets/Icons/Help'
+
+export const TooltipQuestion = props => {
+  return (
+    <Tooltip {...props.props}>
```
Comment (damassi) [inline src/desktop/lib/global_react_modules.js]: This is still a bit strange to me -- where is the key `props` coming from?

File: src/desktop/lib/global_react_modules.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (damassi) [reply src/desktop/lib/global_react_modules.js]: Ah this this makes sense now 👍 

At some point once the pattern has settled a bit we should move these components / wrappers into their own files in force but for now AOK

File: src/desktop/apps/artwork/components/commercial/templates/price.jade [hunk 8]
```diff
@@ -8,11 +8,9 @@ if artwork.sale_message || isPermanentCollection
...
         .artwork-commerical__price
           | #{artwork.sale_message}
-          span(
-            class= artwork.is_price_range ? 'help-tooltip' : ''
-            data-message= 'The range is an approximate indication of the work’s price point, and the exact price is quoted upon request.'
-            data-anchor= 'bottom-right'
-          )
+          if artwork.is_price_range
+            .tooltip-question-container
+              != stitch.components.TooltipQuestion({horizontalAlign: 'left', message: 'The range is an approximate indication of the work’s price point, and the exact price is quoted upon request.'})
```
Comment (oxaudo) [inline src/desktop/apps/artwork/components/commercial/templates/price.jade]: @damassi - here is where I'm using the component exported in `global_react_modules.js`. So props are passed here. Maybe I'm not using it correctly?

Comment (damassi) [reply src/desktop/apps/artwork/components/commercial/templates/price.jade]: This surfaced a pretty interesting edge-case that i was able to fix due to the inability to accurately test around `setTimeout` in jest. Published a new version, incoming to master

Comment (damassi) [reply src/desktop/apps/artwork/components/commercial/templates/price.jade]: Fixed in Force #master

**Comment by oxaudo** [comment]
@damassi - Whatever you did fixed the issue of passing props directly! Thanks:)

@alloy - I think this should be ready to go now.

**Comment by oxaudo** [comment]
I guess part of it was already merged thanks to @damassi trying to resolve the `props` issue here. So the stuff that is left here is just to replace current help icon for price range on artwork with the newly introduced one from react.
@alloy - I'll merge on green unless you have any objections and bring all of that into our https://force-staging-pr-2357.herokuapp.com/ app together with final modifications for hiding the card icon.

**Comment by alloy** [comment]
@oxaudo Sounds great to me 👍
